application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/TrialLicenseGenerator.java

```diff
@@ -250,4 +249,18 @@ private Boolean isOwnerDataComplete()
             || StringUtils.isEmpty(licensingConfig.getLicensingOwnerFirstName())
             || StringUtils.isEmpty(licensingConfig.getLicensingOwnerEmail()));
     }
+
+    /**
+     * Check if the given paid extension has already a license. We need to retrieve license updates from the XWiki Store
+     * before, in case there is already an active license of the extension.
+     *
+     * @param extensionId the extension to be checked
+     * @return true if the extension is licensed, false otherwise.
+     */
+    private Boolean isLicensed(ExtensionId extensionId)
+    {
+        updateLicenses();
```

Comment:
Will this slow down the extension install/upgrade? I hope it doesn't break the install/upgrade when the XWiki Store is down. What I don't like is that:
* you update all licenses: maybe this is needed because the installed extension could be covered by a license generated for another extension, in which case a comment would be good
* you can make this call multiple times during an install if the installed extension has licensed dependencies (which will be installed first); I'm stating to think that listening to ``ExtensionInstalled`` is not the best option, it would be better to listen to "extension install job finished" and then update all licenses (once) and then look for extensions that are missing a license and try to generate one.